accepted/00000-use-dnf-for-rpm-repos.md

```diff
@@ -0,0 +1,175 @@
+issue in SUSE Linux Enterprise, as it's maintained in PackageHub. For the downstream SUSE Manager product, it is
+likely that the DNF stack packages would need to be imported to be maintained exclusively for being used with SUSE Manager.
+
+However, this is unlikely to be much of a maintenance burden as the community packaging is actively maintained
+and well-structured. The packager (who is also the author of this RFC) is responsive to Uyuni/SUSE Manager needs
+and the upstream DNF developers are interested and responsive to community consumers' needs.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Revert back to using YUM
+
+One alternative to resolve this issue is to roll back to using YUM. However, this is a bad idea.
+YUM is dead, and DNF is the maintained successor. YUM does not support Python 3, and there is an
+immediate goal to be able to run Uyuni on a system lacking Python 2.
+
+And even with reverting back to YUM, Uyuni would not be able to properly support RH/Fedora modules.
+Even Spacewalk (which has not yet migrated to DNF) is not able to deal with modules properly as
+they are offered in Red Hat Enterprise Linux 8.
+
+Moreover, [YUM has been retired in Fedora 31](https://fedoraproject.org/wiki/Changes/Retire_YUM_3)
+and "YUM" in Red Hat Enterprise Linux 8 is just a CLI shim to DNF, and only the DNF Python interface
+and libraries are available. Thus, reverting to the "YUM" version would not work.
+
+## Package and port Zypper to other distributions for Uyuni
+
+Another alternative is to package Zypper for distributions that lack it. But this is not appropriate
+for CentOS/RHEL as doing so requires replacing and upgrading packages in the distribution base to
+build it. Notably, the `libsolv` package is too old in CentOS/RHEL, and even if it was not, it is
+not configured in a manner that is usable for the Zypper stack to build on top of it.
+
+For people who wish to use Uyuni on Red Hat Enterprise Linux, a requirement to replace base packages
+is unacceptable, since it would break the terms of supportability per the Red Hat Enterprise Linux
+subscription. There is also the potential consequence of breaking other parts of the system by using
+a different configuration in libsolv than what Red Hat ships.
+
+Finally, Zypper does not implement support for handling RH/Fedora modules, so it will be very hard
+to properly support handling RH/Fedora modules without using DNF anyway (most of the sorting, filtering,
+and solution management logic for modules is implemented in DNF itself right now). This becomes a
+major issue with Red Hat Enterprise Linux 8 as managing it does not work in large part due to this.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+- Will we need to extend libdnf/DNF API to pull arbitrary repodata types mentioned in `repomd.xml`?
+- Would we need to extend the librepo API to pull extra or legacy repodata information?
```

Comment:
If we need to pull stuff not referenced in `repomd.xml`, we probably need to add it to librepo unless we want to just call urlgrabber to fetch it manually...